src/reducers/tracks.js

```diff
@@ -8,8 +8,24 @@ const initialState = {
 export default function tracks(state = initialState, action) {
   switch (action.type) {
     case actionTypes.TRACK_CHANGED:
-      state.previousTracks.push(state.currentTrack);
-      return { ...state, currentTrack: action.track };
+      if (state.currentTrack) {
```

Comment:
this always evaluates to true, can we include it in the returned state? notes below